src/Pulsar.Client/Api/Producer.fs

```diff
@@ -39,34 +42,73 @@ type Producer private (producerConfig: ProducerConfiguration, lookup: BinaryLook
+
                             while pendingMessages.Count > 0 do
-                                let payload = pendingMessages.Dequeue()
-                                do! clientCnx.Send payload
+                                let pendingMessage = pendingMessages.Dequeue()
+                                do this.Mb.PostAndAsyncReply(fun channel ->
+                                    SendMessage (
+                                        pendingMessage.SequenceId,
+                                        pendingMessage.Payload,
+                                        channel)) |> ignore
                                 sentCount <- sentCount + 1
+
                             Log.Logger.LogInformation("{0} pending messages was sent", sentCount)
                     | _ ->
                         Log.Logger.LogWarning("Connection opened but connection is not ready")
                 | ProducerMessage.ConnectionClosed ->
                     do! connectionHandler.ConnectionClosed()
-                | ProducerMessage.SendMessage (payload, channel) ->
+                | ProducerMessage.SendMessage (sequenceId, payload, channel) ->
+
+                    if pendingMessages.Count <= producerConfig.MaxPendingMessages
+                    then
+                        let pendingMessage = { SequenceId = sequenceId; Payload = payload }
+                        pendingMessages.Enqueue(pendingMessage)
+                    else
+                        raise <| ProducerException("Producer send queue is full.")
+
                     match connectionHandler.ConnectionState with
                     | Ready clientCnx ->
                         do! clientCnx.Send payload
                     | _ ->
                         Log.Logger.LogWarning("NotConnected, skipping send")
-                        if pendingMessages.Count < producerConfig.MaxPendingMessages
-                        then
-                            pendingMessages.Enqueue payload
-                        else
-                            raise <| ProducerException("Producer send queue is full.")
+
                     channel.Reply()
                 | ProducerMessage.SendReceipt receipt ->
+
                     let sequenceId = %receipt.SequenceId
-                    match messages.TryGetValue(sequenceId) with
-                    | true, tsc ->
-                        tsc.SetResult(MessageId.FromMessageIdData(receipt.MessageId))
-                        messages.TryRemove(sequenceId) |> ignore
-                    | false, _ -> ()
+                    let pendingMessage = pendingMessages.Peek()
+                    let expectedSequenceId = pendingMessage.SequenceId
+
+                    if sequenceId > expectedSequenceId then
+                        Log.Logger.LogWarning(
+                            "[{0}] [{1}] Got ack for message. Expecting: {2} - got: {3} - queue-size: {4}",
+                            producerConfig.Topic,
```

Comment:
Can you please pass parameters in one line here and below, otherwise the code becomes too long

Reply:
Done